db/migrate/20230831205941_create_pet_applications.rb

```diff
@@ -0,0 +1,11 @@
+class CreatePetApplications < ActiveRecord::Migration[7.0]
+  def change
+    create_table :pet_applications do |t|
+      t.references :pet, null: false, foreign_key: true
+      t.references :application, null: false, foreign_key: true
+      t.string :pet_app_status 
```

Comment:
I don't think that you would need to add this column here. Your applications table has a status on it. So you would just need to use ActiveRecord to reach through the `pet_applications` table 
to get to the status attribute on your applications table. You can do this because 
```
a pet has_many :pet_applications
and has_many :applications, through: :pet_applications
```